pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientCnx.java

```diff
@@ -105,6 +107,9 @@
 
     private final ConcurrentLongHashMap<CompletableFuture<Optional<SchemaInfo>>> pendingGetSchemaRequests = new ConcurrentLongHashMap<>(
             16, 1);
+    // transaction requests that waiting in client side
+    private final ConcurrentLongHashMap<CompletableFuture<TransactionResponse>> pendingGetTransactionRequests =
```

Comment:
make it optional. initialize it lazily so there is no overhead for normal clients.